### ISSUE TITLE
Plugin slows down site loading time when used in pair with Polylang plugin

### ISSUE THREAD

**Issue by gpluta** [iworks/sierotki#1]
I have go this issue, that when I activate Sierotki plugin on my multilang website (I am using Polylang), the website slows down significantly, but only on the translated parts of the website. For intstance it takes one second to load https://www.example.com site, but it takes UP TO A MINUTE to load https://www.example.com/it (the italian version of the main page). Same happens with english pages.

I have noticed this behaviour on a few of my Polylang + Sierotki websites, so it is not a single case.

I have worked around this issue by writing a simple mu-plugin that disables Sierotki on my non-polish pages. This is reasonable, as I do not expect it to work on non-polish websites, but this feels a bit hacky, as I have to parse the url to detect the translation, and Polylang does not have to change the url to actually display a translated site.

I have tried to analyse the code of Sierotki, but failed to find a reason for such behavior.

As I love this plugin and use it on every single of my websites, I would love to help somehow. I am not an experienced WP plugin developer, so I am kind of stuck. All I can do is ask the author to perhaps look into this problem a bit?

Cheers!

**Comment by iworks** [comment]
Hi @gpluta 

Thank you for pointing out this issue - I will try to handle this.

To be honest, I barely use the Polylang, I have a lifetime licence for another translation plugin :D

Marcin

**Comment by iworks** [comment]
@gpluta how it works now with 3.1x?

**Comment by cold2425** [comment]
@iworks I have the same problem as @gpluta and unfortunately the update doesn't solve it.

**Comment by elmaciaso** [comment]
I also have this problem on several pages. I had to turn off the plugin, because it generate so big delay in loading site. 

P.S. Marcinie - Twoja wtyczka jest jedyną która radzi sobie z językiem polski. Szkoda by było z niej rezygnować dlatego, że powoduje takie problemy przy angielskiej wersji strony. Czy planujesz rozwiązać ten błąd?

**Comment by iworks** [comment]
@elmaciaso Yes, I was checking it, and it is an issue with the Polylang plugin, it works like a charm with WPML. I even know what cause a problem - when we have both PL and EN items on a menu.

And it is hard to solve if the Polylang plugin do smth strange.
